kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -288,6 +307,60 @@ private BigQuery getBigQuery() {
     return bigQuery.updateAndGet(bq -> bq != null ? bq : newBigQuery());
   }
 
+  private void setTimePartitioningForTimestamp(PartitionedTableId.Builder builder, TimePartitioning timePartitioning,
+                                               Long timestamp) {
+    if (timePartitioning != null) {
+      switch (timePartitioning.getType()) {
+        case HOUR:
+          builder.setHourPartition(timestamp);
+          break;
+        case MONTH:
+          builder.setMonthPartition(timestamp);
+          break;
+        case YEAR:
+          builder.setYearPartition(timestamp);
+          break;
+        default:
+          builder.setDayPartition(timestamp);
+      }
+    } else {
+      builder.setDayPartition(timestamp);
+    }
+  }
+
+  private void setTimePartitioning(PartitionedTableId.Builder builder, TimePartitioning timePartitioning) {
+    if (timePartitioning != null) {
+      switch (timePartitioning.getType()) {
+        case HOUR:
+          builder.setHourPartitionNow();
+          break;
+        case MONTH:
+          builder.setMonthPartitionForNow();
+          break;
+        case YEAR:
+          builder.setYearPartitionForNow();
+          break;
+        default:
+          builder.setDayPartitionForNow();
+      }
+    } else {
+      builder.setDayPartitionForNow();
+    }
+  }
+
+  private Table retrieveCachedTable(TableId tableId) {
+    String key = tableId.getProject() + "." + tableId.getDataset() + "." + tableId.getTable();
+    Cache<String, Table> cache = getCache(config);
+    Table table = cache.getIfPresent(key);
+
+    if (table == null){
+      table = getBigQuery().getTable(tableId);
+      if (table != null) cache.put(key, table);
+    }
+
+    return table;
+  }
```

Comment:
Discussed elsewhere, just want to comment here for the record--I think the logic here is fine but we should replace use of the third-party `Cache` class with a simple `Map<String, Table>`.